src/meta/typesmap.go

```diff
@@ -30,16 +30,30 @@ const (
 	mapPrecise
 )
 
+type RawTypesMap map[Type]struct{}
```

Comment:
Zero value cannot be used directly, you better provide the New() function in this case.